squidb/src/com/yahoo/squidb/sql/Query.java

```diff
@@ -496,18 +495,15 @@ private void visitSelectClause(SqlBuilder builder, boolean forSqlValidation) {
             builder.sql.append("DISTINCT ");
         }
 
+        List<Field<?>> toSelect;
         if (isEmpty(fields)) {
-            // Explicitly add rowid for virtual tables: "select table.rowid as rowid, *"
-            if (table instanceof VirtualTable) {
```

Comment:
It was originally because "select *" doesn't include columns named rowid. It's safe to remove because now we explicitly inline the rowid field rather than using *. Also there should be a unit test making sure that fetching rowids in the select all case works for virtual table models.